Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForestState.java

```diff
@@ -15,55 +15,54 @@
 
 package com.amazon.randomcutforest;
 
-import java.util.ArrayList;
 import java.util.List;
 
+import lombok.Data;
+
 import com.amazon.randomcutforest.executor.Sequential;
-import com.amazon.randomcutforest.sampler.CompactSamplerData;
 import com.amazon.randomcutforest.sampler.Weighted;
-import com.amazon.randomcutforest.store.PointStoreDoubleData;
-import com.amazon.randomcutforest.tree.TreeData;
+import com.amazon.randomcutforest.state.sampler.CompactSamplerState;
+import com.amazon.randomcutforest.state.store.PointStoreDoubleState;
+import com.amazon.randomcutforest.tree.CompactRandomCutTreeState;
 
 /**
  * A class that encapsulates most of the data used in a RandomCutForest such
  * that the forest can be serialized and deserialized.
  */
-public class ForestState {
-
-    public String typeOfForest = null;
+@Data
+public class RandomCutForestState {
 
-    public long entreesSeen;
+    private long entreesSeen;
 
-    public double lambda;
+    private double lambda;
 
-    public int numberOfTrees;
+    private int numberOfTrees;
 
-    public int sampleSize;
+    private int sampleSize;
 
-    public int dimensions;
+    private int dimensions;
 
-    public int outputAfter;
+    private int outputAfter;
 
-    public boolean storeSequenceIndexesEnabled;
+    private boolean storeSequenceIndexesEnabled;
```

Comment:
raise this for final decision

Reply:
I'd keep this for now. a) if does help some of the inspection/explanation, b) as you point out, ion this information is available and we are not saving tree data, we can use this information to rebuild trees.